app/scripts/modules/core/src/pipeline/config/stages/travis/travisExecutionDetails.controller.ts

```diff
@@ -21,6 +21,13 @@ export class TravisExecutionDetailsCtrl implements IController {
     this.stage = this.$scope.stage;
     this.initialize();
     this.$scope.$on('$stateChangeSuccess', () => this.initialize());
+    this.$scope.$watch(
+      'stage',
+      function() {
+        this.stage = $scope.stage;
+      }.bind(this),
+      true,
+    );
```

Comment:
```suggestion
    this.$scope.$watch('stage.refId', () => this.stage = $scope.stage);
```

What are your thoughts on simplifying to something like the above? With an arrow function, we can avoid binding `this` since it will use the `this` value of its enclosing lexical scope. Additionally, if we watch the `refId` rather than the entire stage object, we can avoid an expensive deep equality check.

Thanks again for stepping up to resolve this bug!

Reply:
Awesome! I've changed it and it works flawlessly

Thanks again for your patience @maggieneterval